### ISSUE TITLE
[Windows] MissingPluginException(No implementation found for method getAll on channel dev.fluttercommunity.plus/package_info)

### ISSUE THREAD

**Issue by Areopagitics** [fluttercommunity/plus_plugins#760]
## System info

Platform the Issue occurs on: WINDOWS
Plugin name: package_info_plus
Plugin version: 1.4.0

## Steps to Reproduce

when calling
`PackageInfo packageInfo = await PackageInfo.fromPlatform();`
it throws
`MissingPluginException(No implementation found for method getAll on channel dev.fluttercommunity.plus/package_info)`
error

<details>
<summary>Logs</summary>

```[   +2 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 
[   +1 ms] R Hot restart.
[   +1 ms] h List all available interactive commands.
[   +1 ms] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ]  Running with sound null safety 
[   +1 ms] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:63075/VLhRXO1aR_A=/
[ +283 ms] [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: MissingPluginException(No implementation found for method getAll on channel dev.fluttercommunity.plus/package_info)
[   +1 ms] #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:175:7)
[        ] <asynchronous suspension>
[        ] #1      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:377:43)
[        ] <asynchronous suspension>
[        ] #2      MethodChannelPackageInfo.getAll (package:package_info_plus_platform_interface/method_channel_package_info.dart:13:17)
[        ] <asynchronous suspension>
[        ] #3      PackageInfo.fromPlatform (package:package_info_plus/package_info_plus.dart:36:26)
[        ] <asynchronous suspension>
[        ] #4      loadSettings (package:myApp/src/helper.dart:113:29)
[        ] <asynchronous suspension>
[   +7 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:63075/VLhRXO1aR_A=/
```

</details>

<details>
<summary>Flutter doctor output</summary>

```
[√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.19043.1526], locale en-US)
    • Flutter version 2.10.2 at C:\Users\LAPTOP\Desktop\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (2 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0-rc1)
    • Android SDK at C:\Android
    • Platform android-31, build-tools 32.0.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.0.32112.339
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 4.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\LAPTOP\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1526]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 98.0.4758.102
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 97.0.1072.55

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by liorboyango** [comment]
Using version 1.2.1 for now, for some reason 1.3.x had the same behavior.
Can't believe I wasted my time with `flutter pub cache repair`, thinking the problem was on my side :(

**Comment by lohanidamodar** [comment]
@liorboyango 1.3.0 works, (1.3.1 and 1.4.0 both give this error even in Linux) for me as well.

**Comment by liorboyango** [comment]
Indeed, edited my previous comment

**Comment by fabio21** [comment]
[Windows ] MissingPluginException(No implementation found for method getAll on channel dev.fluttercommunity.plus/package_info)

version: package_info_plus: ^1.3.0

**Comment by Areopagitics** [comment]
Thanks for the feedback! I don't understand why in the _method_channel_package_info.dart_ file on line 7 it still refers to _package_info_ if it's deprecated.
```
const MethodChannel _channel =
    MethodChannel('dev.fluttercommunity.plus/package_info');
```

**Comment by LiveRock** [comment]
I am having this bug too. using version 1.3.0 get rid of the exception but am unable to get the version and build number of the app.
Building for Windows.

flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.10.2, on Microsoft Windows [Version 10.0.22000.527], locale en-SG)
[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.4)
[!] Android Studio (not installed)
[√] VS Code (version 1.64.2)
[√] Connected device (3 available)
[√] HTTP Host Availability

! Doctor found issues in 2 categories.

**Comment by jing-pei** [comment]
same issue

**Comment by fabio21** [comment]
change ^1.3.0 for 1.3.0
and remove pubspec.lock 

flutter pub get

**Comment by gslender** [comment]
The issue is because the windows 1.04 package has the old version of the code still. The github repo has the fixes real implementation, but for some reason the release from 4 months ago is still the stub vs real version. If the owner could update the release then we'd have this fixed.

**Comment by ibrahimlee** [comment]
On Windows use `PackageInfoData packageInfo = await PackageInfoWindows().getAll();` instead of `PackageInfo packageInfo = await PackageInfo.fromPlatform();`

**Comment by Areopagitics** [comment]
> The issue is because the windows 1.04 package has the old version of the code still. The github repo has the fixes real implementation, but for some reason the release from 4 months ago is still the stub vs real version. If the owner could update the release then we'd have this fixed.

yeah, this seems to be a problem with multiple official repos; it's starting to become the biggest pain with Flutter

**Comment by anaisbetts** [comment]
Still broken in v1.4.0

**Comment by gslender** [comment]
The issue is that \package_info_plus_windows-1.0.4 in pub-cache contains package_info_plus_windows_stub in the \lib folder of that windows package. So the way the update was built failed to include the updated windows code that is available on github. Unless the maintainers redo the publishing to pub.dev, the only solution is to manually copy the windows package into your project as an override dependancy.

Just annoying that we can't get any of the maintainers to pay attention and fix this - because it doesn't need any code changes, just a repackage of the correct libraries into a release.

**Comment by jmatth** [comment]
TL;DR fix for anyone arriving here from google: put the following in your `pubspec.yaml`

```yaml
dependency_overrides:
  package_info_plus_windows:
    git:
      url: https://github.com/fluttercommunity/plus_plugins.git
      ref: main
      path: packages/package_info_plus/package_info_plus_windows
```

**Comment by litang0908** [comment]
> TL;DR fix for anyone arriving here from google: put the following in your `pubspec.yaml`
> 
> ```yaml
> dependency_overrides:
>   package_info_plus_windows:
>     git:
>       url: https://github.com/fluttercommunity/plus_plugins.git
>       ref: main
>       path: packages/package_info_plus/package_info_plus_windows
> ```

Thank your, it works!
